### ISSUE TITLE
Kanban View "Create" button is hidden even after adding a "Select" field

### ISSUE THREAD

**Issue by JarWarren** [twentyhq/twenty#5915]
## Steps to Reproduce
- Create a new custom object
- Add a "Select" field
- Attempt to create a Kanban view
- The button is missing

<img width="234" alt="Screenshot 2024-06-17 at 9 13 00 PM" src="https://github.com/twentyhq/twenty/assets/43893126/4d27d48e-ed8f-4d73-9abc-7cbc05c8f773">

## Expected behavior

There should always be a CTA button.
- No "Select" field: "Go to Settings"
- Otherwise: "Create"

## Technical inputs

The workaround is to first, create a second Table view. Then, for some reason, you will be allowed to create a Kanban.

**Comment by JarWarren** [comment]
I can take a look at this on the weekend. If anyone wants it before then, they're welcome to have it though. It's probably a good first issue.

**Comment by Us3r-gitHub** [comment]
Let me attempt to develop it.

I would like to discuss it after completing the local installation setup.

**Comment by Us3r-gitHub** [comment]
I've traced the issue.

TLDR: It happens because KanbanMetadataId is not being set properly.

Upon returning from the settings page (after adding a new select field there), it must update the dirty state of the `ViewPicker` to trigger an effect that sets the `KanbanMetadataId`. [code](https://github.com/twentyhq/twenty/blob/main/packages/twenty-front/src/modules/views/view-picker/components/ViewPickerCreateOrEditContentEffect.tsx#L68)

Do you have any ideas on how to achieve this?
Alternatively, are there any other solutions we could consider?
